### PR TITLE
feat: Add `requestComplexity.allowRegex` option to disable `$regex` query operator

### PR DESCRIPTION
**Pull request by mtrezza**
## Issue

Add option to disable regex query operator

## Tasks

- [x] Add new feature
- [x] Add tests
- [x] CI passes

## Summary by CodeRabbit

* **New Features**
  * Added an optional allowRegex setting (default: true) to control whether non-master-key requests may use the $regex query operator; master and maintenance keys still bypass the restriction.  

* **Bug Fixes**
  * Rejects $regex usage when disabled, including nested $or/$and/$nor and empty-string $regex; LiveQuery subscriptions are validated similarly.

* **Documentation**
  * Updated configuration docs for allowRegex.

* **Tests**
  * Added comprehensive REST and LiveQuery tests covering allowRegex behavior and config validation.

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -160,6 +160,9 @@ const validateQuery = (
 
   Object.keys(query).forEach(key => {
     if (query && query[key] && query[key].$regex) {
+      if (!isMaster && rc && rc.allowRegex === false) {
+        throw new Parse.Error(Parse.Error.INVALID_QUERY, '$regex operator is not allowed');
+      }
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply src/Controllers/DatabaseController.js:165]: Fixed. Changed the truthiness check to `$regex !== undefined` to properly handle empty-string patterns. Added a test for the edge case.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.8.0-alpha.11](https://github.com/parse-community/parse-server/releases/tag/9.8.0-alpha.11)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.8.0](https://github.com/parse-community/parse-server/releases/tag/9.8.0)
